bika/lims/content/sample.py

```diff
@@ -859,6 +859,34 @@ def getSamplePartitions(self):
         partitions = self.objectValues('SamplePartition')
         return partitions
 
+    def getSamplingDeviationTitle(self):
+        if self.getSamplingDeviation():
+            return self.getSamplingDeviation().Title()
+        return ''
+
+    def getStorageLocationTitle(self):
+        if self.getStorageLocation():
+            return self.getStorageLocation().Title()
+        return ''
+
+    def getSamplePointTitle(self):
+        if self.getSamplePoint():
+            return self.getSamplePoint().Title()
+        return ''
+
+    def getObjectWorkflowStates(self):
+        """
+        Returns a dictionary with the workflow id as key and workflow state as
+        value.
+        :returns: {'review_state':'active',...}
+        """
+        workflow = getToolByName(self, 'portal_workflow')
+        states = {}
+        for w in workflow.getWorkflowsFor(self):
+            state = w._getWorkflowStateOf(self).id
+            states[w.state_var] = state
+        return states
+
```

Comment:
is `getObjectWorkflowStates` used somewhere?
If so, always better to use the functions available at `bika.lims.workflow`. For the retrieval of the current state of an object (for `review_state` id), you can use the function `bika.lims.workflow.getCurrentState`:
https://github.com/naralabs/bika.lims/blob/d482457b6b260c45c5d6862111cf9e35f50a058c/bika/lims/workflow/__init__.py#L366-L371
If you still need a dictionary with the states for all associated workflows to this Sample, consider to add this function in `bika.lims.workflow.__init__.py` (a good signature name could be `getCurrentStates`).

Reply:
```forlderitems``` method from ```SamplesView``` class calls its parent class (```BikaListingView```) ```folderitems``` method. 

https://github.com/naralabs/bika.lims/blob/b14192f19a0771a1af9f7f7b9a67096ab656b5f7/bika/lims/browser/sample/view.py#L505

Taking a look at ```BikaListingView```'s ```folderitems``` method shows that it is accessing and retrieving the value of the column ```getObjectWorkflowStates```.  

https://github.com/naralabs/bika.lims/blob/b14192f19a0771a1af9f7f7b9a67096ab656b5f7/bika/lims/browser/bika_listing.py#L967-L968

```getObjectWorkflowStates``` is also listed as a column in catalog_basic_template.py

https://github.com/naralabs/bika.lims/blob/b14192f19a0771a1af9f7f7b9a67096ab656b5f7/bika/lims/catalog/catalog_basic_template.py#L33-L46